android/app/src/main/kotlin/net/mullvad/mullvadvpn/viewmodel/ConnectViewModel.kt

```diff
@@ -133,6 +133,13 @@ class ConnectViewModel(
             .debounce(UI_STATE_DEBOUNCE_DURATION_MILLIS)
             .stateIn(viewModelScope, SharingStarted.WhileSubscribed(), ConnectUiState.INITIAL)
 
+    init {
+        viewModelScope.launch {
+            outOfTimeUseCase().filter { it }.first()
```

Comment:
I'm slightly confused here, would this trigger while the user is using the app or only they navigate to this screen and why this change in behaviour then?

Reply:
We will launch it as soon as they open the connect screen. If we find a outOfTime event that return true (they are then out of time), this launch will finish and we will navigate to the OutOfTimeScreen. Previously this was done in the collection of UIState but this is faulty due to it can cause us to call navigate multiple times as we derive a new state..

Reply:
Yes, but if we happen to be out of time when we are on the ConnectScreen? Will we show out of time then?

Reply:
Ok looked at the code and got it now. Maybe instead of `invoke` maybe we should call it something like `isOutOfTime` ?